src/Findologic/Config/FindologicConfigService.php

```diff
@@ -0,0 +1,307 @@
+        if (!\is_array($value)) {
+            return (int)$value;
+        }
+
+        throw new InvalidSettingValueException($key, 'int', \gettype($value));
+    }
+
+    public function getFloat(string $key, string $salesChannelId, string $languageId): float
+    {
+        $value = $this->get($key, $salesChannelId, $languageId);
+        if (!\is_array($value)) {
+            return (float)$value;
+        }
+
+        throw new InvalidSettingValueException($key, 'float', \gettype($value));
+    }
+
+    public function getBool(string $key, string $salesChannelId, string $languageId): bool
+    {
+        return (bool)$this->get($key, $salesChannelId, $languageId);
+    }
+
+    public function all(string $salesChannelId, string $languageId): array
+    {
+        return $this->load($salesChannelId, $languageId);
+    }
+
+    /**
+     * @throws InvalidDomainException
+     * @throws InvalidUuidException
+     * @throws InconsistentCriteriaIdsException
+     */
+    public function getDomain(string $salesChannelId, string $languageId): array
+    {
+        $domain = 'FinSearch';
+        $queryBuilder = $this->connection->createQueryBuilder()
+            ->select('LOWER(HEX(id))')
+            ->from('finsearch_config')
+            ->where('sales_channel_id = :salesChannelId')
+            ->andWhere('language_id = :languageId');
+
+        $domain = rtrim($domain, '.') . '.';
+        $escapedDomain = str_replace('%', '\\%', $domain);
+
+        $salesChannelId = Uuid::fromHexToBytes($salesChannelId);
+        $languageId = Uuid::fromHexToBytes($languageId);
+
+        $queryBuilder->andWhere('configuration_key LIKE :prefix')
+            ->orderBy('configuration_key', 'ASC')
+            ->addOrderBy('sales_channel_id', 'ASC')
+            ->setParameter('prefix', $escapedDomain . '%')
+            ->setParameter('salesChannelId', $salesChannelId)
+            ->setParameter('languageId', $languageId);
+        $ids = $queryBuilder->execute()->fetchAll(FetchMode::COLUMN);
+
+        if (empty($ids)) {
+            return [];
+        }
+
+        $criteria = new Criteria($ids);
+        /** @var SystemConfigCollection $collection */
+        $collection = $this->finSearchConfigRepository
+            ->search($criteria, Context::createDefaultContext())
+            ->getEntities();
+
+        $collection->sortByIdArray($ids);
+        $merged = [];
+        foreach ($collection as $cur) {
+            if (!array_key_exists($cur->getConfigurationKey(), $merged) || !empty($cur->getConfigurationValue())) {
+                $merged[$cur->getConfigurationKey()] = $cur->getConfigurationValue();
+            }
+        }
+
+        return $merged;
+    }
+
+    /**
+     * @param array|bool|float|int|string|null $value
+     */
+    public function set(string $key, $value, string $salesChannelId, string $languageId): void
+    {
+        // reset internal cache
+        $this->configs = [];
```

Comment:
What are you resetting? Why are you resetting? Why needs it to be done here? Shouldn't it be replaced afterwards? That's an odd side effect.

Reply:
This is how shopware handles it. It stores the configuration in memory, and it refreshes it when you save a configuration.

Reply:
That comment does not make much sense then. Refer to a method which does this or call that method explicitly. Also why are you clearing the configuration, instead of overriding it with the new values?